### PR TITLE
{phys}[foss/2020b,intel/2020b] Octopus v11.2

### PR DESCRIPTION
**Pull request by micaeljtoliveira**

### CODE REVIEW

**Comment by migueldiascosta** [comment]
> @smoors It seems those libraries require FFTW, but are incompatible with the MKL implementation of FFTW.

@micaeljtoliveira why not simply add `FFTW` as a dependency? it's not incompatible with the `intel` toolchain

**Comment by micaeljtoliveira** [comment]
> @micaeljtoliveira why not simply add `FFTW` as a dependency? it's not incompatible with the `intel` toolchain

@migueldiascosta My understanding (and previous experience) is that if you link against both MKL and FFTW (e.g., if you need BLAS/LAPACK from MKL and FFT's from FFTW), then at link time the FFTW symbols will be picked from MKL, not FFTW.

**Comment by migueldiascosta** [comment]
> > @micaeljtoliveira why not simply add `FFTW` as a dependency? it's not incompatible with the `intel` toolchain
> 
> @migueldiascosta My understanding (and previous experience) is that if you link against both MKL and FFTW (e.g., if you need BLAS/LAPACK from MKL and FFT's from FFTW), then at link time the FFTW symbols will be picked from MKL, not FFTW.

that's strange, because the MKL symbols for FFTW are in separate libraries `libfftw...` (separately built from the `interfaces` folder), and the FFT symbols in `libmkl...` are not for FFTW but for Intel's DFTI...

**Comment by micaeljtoliveira** [comment]
> > > @micaeljtoliveira why not simply add `FFTW` as a dependency? it's not incompatible with the `intel` toolchain
> > 
> > 
> > @migueldiascosta My understanding (and previous experience) is that if you link against both MKL and FFTW (e.g., if you need BLAS/LAPACK from MKL and FFT's from FFTW), then at link time the FFTW symbols will be picked from MKL, not FFTW.
> 
> that's strange, because the MKL symbols for FFTW are in separate libraries `libfftw...` (separately built from the `interfaces` folder), and the FFT symbols in `libmkl...` are not for FFTW but for Intel's DFTI...

Ah, interesting. Looking into my EasyBuild installation of MKL I see that's indeed the case, but I'm quite sure this is not the case in other installations of MKL I've seen elsewhere. Is this an option of the MKL installer that EasyBuild uses?

Anyway, if that's the case I can try adding FFTW as dependency and see what happens.

**Comment by migueldiascosta** [comment]
> Ah, interesting. Looking into my EasyBuild installation of MKL I see that's indeed the case, but I'm quite sure this is not the case in other installations of MKL I've seen elsewhere. Is this an option of the MKL installer that EasyBuild uses?

If I remember correctly, without EasyBuild the FFTW interfaces didn't even get built by default, we had to do that "manually" (but it's been a while since I last installed MKL without EasyBuild, maybe something has changed in the meantime)

The `imkl` easyblock does build them (unless the easyconfig sets `interfaces` to `False`)

**Comment by micaeljtoliveira** [comment]
@migueldiascosta I finally had the time to test PFFT and libvdwxc with the intel toolchain and it seems to work fine.

File: easybuild/easyconfigs/o/Octopus/Octopus-11.2-intel-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+configopts += 'FCFLAGS_FFTW="-I${EBROOTIMKL}/mkl/include/fftw" '
+configopts += '--with-libxc-prefix=$EBROOTLIBXC '
+configopts += '--with-netcdf-prefix=$EBROOTNETCDFMINFORTRAN '
+configopts += '--with-nlopt-prefix=$EBROOTNLOPT '
+configopts += '--with-futile-prefix=$EBROOTFUTILE '
+configopts += '--with-psolver-prefix=$EBROOTPSOLVER '
+
+# Run only the short tests, as the full testsuite can take a lot of time.
+# Use only two threads as the tests are not large enough to exploit OpenMP parallelism efficiently.
+runtest = 'OMP_NUM_THREADS=2 MPIEXEC="`which mpirun` --bind-to none" check-short'
```
[the remark this answers is not in the document]
Comment (micaeljtoliveira) [reply easybuild/easyconfigs/o/Octopus/Octopus-11.2-intel-2020b.eb]: @smoors Unfortunately I had to change this further. The problem is that `%(mpi_cmd_prefix)s` adds `-n 1`, which then conflicts with the setting of number of processes done by the Octopus test script. I also added a patch to remove the not very nice use of `which`.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3122.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/bf4cca3fa66652092fbb6231b12ff8b4 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by boegel** [comment]
closing this since `foss/2020b` is deprecated, see https://docs.easybuild.io/policies/toolchains

see also:
* #15167

**Pull request closed without merging**